matching/README.md

```diff
@@ -58,9 +58,18 @@ syntax Pattern ::= Variable // variable patterns
                  | "Map" [(Pattern, Pattern)] Variable // map patterns with a variable
 ```
 
+Variables come in two varieties: anonymous variables, which do not get
+added to the substitution, and named variables, which do get added to the
+substitution. Named variables, when matched, are bound to a particular
+occurrence, which corresponds to a term at runtime. Anonymous variables match
+any pattern, like a named variable, but are not bound to anything.
+
 Note that one restriction not captured by the above grammar is that List, Set,
 and Map patterns with a Variable must have at least one element Pattern.
-Otherwise, they would simply be a variable pattern.
+Otherwise, they would simply be a variable pattern. The reason that `Set [] V`
+is equal to `V` is that it can appear only in a context that expects a set
+sort, which means that `List`, `Map`, or regular constructors cannot appear.
+The same, obviously, is true respectively for lists and maps.
```

Comment:
This makes sense; I hadn't appreciated the sortedness constraint on where the constructors appear.